dataprofiler/tests/profilers/utils.py

```diff
@@ -179,19 +179,28 @@ def assert_profiles_equal(actual, expected):
     :param expected: profile to compare to actual
     :type expected: instance of BaseProfiler or BaseColumnProfiler
     """
-    actual_dict = actual.__dict__
-    expected_dict = expected.__dict__
+    actual_dict = actual.__dict__ if not isinstance(actual, dict) else actual
+    expected_dict = expected.__dict__ if not isinstance(expected, dict) else expected
 
     assert len(actual_dict.keys()) == len(expected_dict.keys())
+    assert actual_dict.keys() == expected_dict.keys()
 
     for actual_value, expected_value in zip(
         actual_dict.values(), expected_dict.values()
     ):
-        assert type(actual_value) == type(expected_value)
+        # Condition to test whether the types are equal when a value can be float or float64
+        if type(actual_value) is np.float64 or type(expected_value) is np.float64:
```

Comment:
Loose type comparison for float and float64 for variables that allow for both types